trinity/plugins/builtin/ethstats/plugin.py

```diff
@@ -0,0 +1,63 @@
+import asyncio
+
+from argparse import (
+    ArgumentParser,
+    _SubParsersAction,
+)
+
+from trinity.extensibility import (
+    BaseEvent,
+    BasePlugin,
+    PluginContext,
+)
+from trinity.extensibility.events import (
+    ResourceAvailableEvent,
+    TrinityStartupEvent,
+)
+from eth.chains.base import (
+    BaseChain
+)
+from p2p.peer import (
+    PeerPool
+)
+
+from trinity.plugins.builtin.ethstats.ethstats_service import (
+    EthstatsService,
+)
+
+
+class EthstatsPlugin(BasePlugin):
+
+    def __init__(self) -> None:
+        self.is_enabled: bool = False
+
+        self.chain: BaseChain = None
+        self.peer_pool: PeerPool = None
```

Comment:
It seems that getting the `len()` of the `PeerPool` is currently the only thing that this plugin needs from the `PeerPool`. There's an alternative way of retrieving the number of peers that works cross process.

https://github.com/ethereum/py-evm/blob/d225f4a26811aa15f450378aa8d618caa573b158/trinity/rpc/modules/net.py#L20-L21

I think we should aim for this plugin to become a `BaseIsolatedPlugin` similar to how the `JSON-RPC Server` works. I don't see that as a definitive requirement to be addressed before merging as the plugin doesn't run by default anyway but I *think* this is something that we should do. There will probably be many things that isolated plugins can't do right now but our cross process support is improving and in the long term most things including stuff like the transaction  pool will most likely end up as isolated plugins.

Here's the `JSON-RPC` server to get you started:

https://github.com/ethereum/py-evm/blob/d225f4a26811aa15f450378aa8d618caa573b158/trinity/plugins/builtin/json_rpc/plugin.py#L30-L70

Reply:
Refactored plugin to be `BaseIsolatedPlugin`, used suggested method to get peer count.